src/components/layout/useMenuState.ts

```diff
@@ -16,45 +13,66 @@ interface MenuState {
 export function useMenuState(defaultOpen = false): MenuState {
   const [isMenuOpen, setMenuOpen] = useState<boolean>(defaultOpen);
 
+  const router = useRouter();
+
+  /*
+   * Retrieve the URL for the current route with menu state,
+   * including query params.
+   */
+  function getFullMenuUrl(): string {
+    let fullMenuUrl = `${router.pathname}?menu`;
+
+    if (router.query.code) {
+      fullMenuUrl = fullMenuUrl.replace('[code]', router.query.code as string);
+    }
+
+    const queryParams = Object.entries(router.query)
+      .filter(([key]) => ['menu', 'code'].indexOf(key) === -1)
```

Comment:
nitpick; I think this is more readable

```suggestion
      .filter(([key]) => ['menu', 'code'].includes(key))
```